build/tools/Automation/Program.cs

```diff
@@ -2,180 +2,46 @@
-                    "2. String repo absolute path to the root of Oryx repo.\n");
+                Console.WriteLine($"\nPlease enter a supported platform (e.g. {string.Join(", ", supportedPlatforms)}) and the Oryx root path.");
                 Environment.Exit(1);
             }
 
-            string dateTarget = args[0];
-            repoAbsolutePath = args[1];
-
-            Console.WriteLine($"[Main] dateTarget: {dateTarget}");
-            await AddNewPlatformConstantsAsync(dateTarget);
-
-            return 0;
-        }
-
-        /// <Summary>
-        /// Adds new platform constants to Oryx repo
-        /// TODO: use dependency injection
-        /// </Summary>
-        public static async Task AddNewPlatformConstantsAsync(string dateTarget)
-        {
-            DotNet dotNet = new DotNet(repoAbsolutePath, prodSdkVersions);
-            await dotNet.PullSdkVersionsAsync(Constants.DotNetName);
-            List<PlatformConstant> platformConstants = await dotNet.GetPlatformConstantsAsync(dateTarget);
-            List<Constant> yamlConstants = await DeserializeConstantsYamlAsync();
-            dotNet.UpdateConstants(platformConstants, yamlConstants);
-            prodSdkVersions.Clear();
-
-            // TODO: add functionality for other platforms (python, java, golang, etc).
-        }
-
-        /// <Summary>
-        /// Updates:
-        ///     - Constants.ConstantsYaml
-        ///
-        /// Deserializes Constants.ConstantsYaml for platforms, that have new releases, can update.
-        /// </Summary>
-        public static async Task<List<Constant>> DeserializeConstantsYamlAsync()
-        {
-            var constantsYamlAbsolutePath = Path.Combine(repoAbsolutePath, "build", Constants.ConstantsYaml);
-            string fileContents = await File.ReadAllTextAsync(constantsYamlAbsolutePath).ConfigureAwait(true);
-
-            var deserializer = new DeserializerBuilder()
-                .WithNamingConvention(UnderscoredNamingConvention.Instance)
-                .Build();
-            var yamlContents = deserializer.Deserialize<List<Constant>>(fileContents);
-            return yamlContents;
-        }
-
-        public static bool DatesMatch(string dateTarget, string dateReleased)
-        {
-            var releasedDate = DateTime.Parse(dateReleased);
-            var targetDate = DateTime.Parse(dateTarget);
-            int datesMatch = DateTime.Compare(releasedDate, targetDate);
-            bool match = datesMatch == 0;
-            Console.WriteLine($"[DatesMatch]  releasedDate: {releasedDate} targetDate: {targetDate} " +
-                $"datesMatch: {datesMatch} match: {match}");
-            return match;
-        }
+            var serviceProvider = new ServiceCollection()
+                .AddHttpClient()
+                .AddSingleton<IVersionService, VersionService>()
+                .AddSingleton<IYamlFileReaderService, YamlFileReaderService>()
+                .AddScoped<DotNet.DotNet>()
```

Comment:
Similar comment in the Python PR, but it would be nice to rename these platform processing classes to something like `DotNetProcessor` or `DotNetAutomator` so that we don't need the redundant namespace suffix and the class name more accurately describes what it's responsible for.

Reply:
Updated